include/swift/AST/DiagnosticsSema.def

```diff
@@ -905,6 +905,8 @@ ERROR(cannot_find_in_scope,none,
 ERROR(cannot_find_in_scope_corrected,none,
       "cannot %select{find|find operator}1 %0 in scope; did you mean '%2'?",
       (DeclNameRef, bool, StringRef))
+ERROR(cannot_find_self_in_scope,none,
+      "cannot find 'self' in scope; did you mean to use it within a type or extension context?", ())
```

Comment:
"Within" can just be "in," but overall I think this looks great 👍

Reply:
There